base/setup/lib/utils/partlist.c

```diff
@@ -3753,26 +3809,85 @@ WritePartitionsToDisk(
         }
     }
 
+    /* The PARTMGR should have notified the MOUNTMGR that new volumes
+     * associated with the new partitions had to be created */
+
+    /* Assign valid device names to new volumes */
+    for (Entry = List->VolumesList.Flink;
+         Entry != &List->VolumesList;
+         Entry = Entry->Flink)
+    {
+        Volume = CONTAINING_RECORD(Entry, VOLENTRY, ListEntry);
+        InitVolumeDeviceName(Volume, NULL);
+    }
+
     return TRUE;
 }
 
-BOOLEAN
+VOID
+SetMBRPartitionType(
+    IN PPARTENTRY PartEntry,
+    IN UCHAR PartitionType)
```

Comment:
In the next function at line 3872 in the same file you used SAL2, but here not. Why?

Reply:
I used SAL2 for functions I modified their implementation; for functions that I just moved around without implementation change (like this one), I didn't change the annotations.

Reply:
ok, resolved.